extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTOTPSecretEngine.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.vault;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.quarkus.vault.secrets.totp.CreateKeyParameters;
+import io.quarkus.vault.secrets.totp.KeyConfiguration;
+import io.quarkus.vault.secrets.totp.KeyDefinition;
+
+/**
+ * This service provides access to the TOTP secret engine.
+ *
+ * @see VaultRuntimeConfig
+ */
+public interface VaultTOTPSecretEngine {
+
+    /**
+     * Creates or updates a key definition.
+     * 
+     * @param name of the key.
+     * @param createKeyParameters required to create or update a key.
+     * @return Barcode and/or URL of the created OTP key.
+     */
+    Optional<KeyDefinition> createKey(String name, CreateKeyParameters createKeyParameters);
```

Comment:
Can the dynamic key operations be of the practical interest to the application code ?

Reply:
This key is a TOTP key with different properties than a standard key, in this case, a key just contains two fields a QR code and a URL.